From 2cfe4c4a750a90767d9e4b0224c1139bde4fc3f1 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Tue, 28 May 2019 13:35:56 +0300
Subject: [PATCH 200/235] sdk: dts: ls1046: drop smmu from the frwy sdk dtses

Drop the smmu from the frwy and frwy-usdpaa versions of the SDK device
trees because SMMU is supported only for the upstream version of the
dpaa ethernet drivers.

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts    | 14 ++++++++++++++
 .../boot/dts/freescale/fsl-ls1046a-frwy-usdpaa.dts | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
index 1df7dfc90c0a..7c9a9f14ab49 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-sdk.dts
@@ -26,6 +26,20 @@
 &soc {
 #include "qoriq-dpaa-eth.dtsi"
 #include "qoriq-fman3-0-6oh.dtsi"
+
+	pcie@3400000 {
+		/delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+		/delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+		/delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 
 &fsldpaa {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-usdpaa.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-usdpaa.dts
index 8e58a0bf3c18..38cbeaec70c3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-frwy-usdpaa.dts
@@ -69,6 +69,20 @@
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
 	};
+
+	pcie@3400000 {
+		/delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+		/delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+		/delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 / {
 	reserved-memory {
-- 
2.17.1

